listenbrainz/db/donation.py

```diff
@@ -0,0 +1,124 @@
+import time
+from typing import Optional
+
+import psycopg2
+import sqlalchemy
+from psycopg2.extras import execute_values
+from sqlalchemy import create_engine, NullPool, text
+
+engine: Optional[sqlalchemy.engine.Engine] = None
+
+
+def init_meb_db_connection(connect_str):
+    """Initializes database connection using the specified Flask app."""
+    global engine
+    while True:
+        try:
+            engine = create_engine(connect_str, poolclass=NullPool)
+            break
+        except psycopg2.OperationalError as e:
+            print("Couldn't establish connection to db: {}".format(str(e)))
```

Comment:
Should this be a logging call, rather than print?

Reply:
The logger is likely not initialized at this, because we are still inside create_app. Other db connection init code uses print for the same reason. I think logging might work but there's a detail or two I need to look up to be sure. for another day.

Reply:
Ah, perhaps just document it then and move on.